ui/main.qml

```diff
@@ -220,6 +225,21 @@ StatusWindow {
         }
     }
 
+    // On MacOS, explicitely restore the window on activating
+    Connections {
+        target: Qt.application
+        enabled: Qt.platform.os === Constants.mac
+        function onStateChanged() {
+
+            if (d.previousApplicationState == Qt.application.state) {
```

Comment:
I can't understand it... so if `d.previousApplicationState == Inactive` and `Qt.application.state == Inactive` as well it should be shown normal? And be visible?

Reply:
oh! good catch!
I was copying the logic from `DockShowEvent`
https://github.com/status-im/status-desktop/blob/bf4af3d7d4c32e0b9ecf71f6d8051b79996fcef0/vendor/DOtherSide/lib/src/Status/DockShowAppEvent.cpp#L27-L28

But lost half of it 😄 

I don't really understand why the state should be the same as previous, but I decided not to break it, as it was working fine.